Bottom: 893863455c1c7c95b764e00fed46ba34cc852a8c
Top:    6d8f7db0320d136475a5ddfaca9e9c5585a12d10
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 16:59:24 +0100

Refresh of draw-c-separate-gathering-rout

---

diff --git a/src/draw.c b/src/draw.c
index 7c84a76..d99b94d 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -518,25 +518,21 @@ draw_pin (PinTypePtr pin)
 static void
 draw_via (PinTypePtr via)
 {
-  SetPVColor (Pin, VIA_TYPE);
+  SetPVColor (via, VIA_TYPE);
   draw_pv (via);
 }
 
 static int
 via_callback (const BoxType * b, void *cl)
 {
-  PinTypePtr via = (PinTypePtr) b;
-  SetPVColor (via, VIA_TYPE);
-  draw_pv (via);
+  draw_pin ((PinType *)b);
   return 1;
 }
 
 static int
 pin_callback (const BoxType * b, void *cl)
 {
-  PinTypePtr pin = (PinTypePtr) b;
-  SetPVColor (pin, PIN_TYPE);
-  draw_pv (pin);
+  draw_via ((PinType *)b);
   return 1;
 }
